vortex-array/src/array/datetime/localdatetime.rs

```diff
@@ -54,6 +54,26 @@ impl LocalDateTimeArray {
     }
 }
 
+impl TryFrom<LocalDateTimeArray> for ExtensionArray {
```

Comment:
this was missing to get from the LocalDateTimeArray back to the ExtensionArray, so we can return a LocalDateTimeArray as a flattened thing